#Top 10 softwares for ChIP-seq data analysis. (The 2nd generation sequencing)
             
## I. Check quality for FASTQ files
                          
## II. Filter and correct the raw reads
                  
## III. Reads mapping (Mappers or Aligners)
  1. BWA-mem, BWA-aln :    https://github.com/lh3/bwa        
  2. Bowtie2 :      https://github.com/BenLangmead/bowtie2   
  3. Stampy :   www.well.ox.ac.uk/bioinformatics/Software/Stampy-latest.tgz   
  4. Novoalign : http://www.novocraft.com/support/download/  
  5. Subread : http://subread.sourceforge.net/     
  6. Yara : https://www.seqan.de/apps/yara/       
     RazerS 3 : https://www.seqan.de/apps/razers-3/      
  7. mrsFAST-Ultra :  http://sfu-compbio.github.io/mrsfast/      
  8. BBMap : https://sourceforge.net/projects/bbmap/            
  9. GMAP-GSNAP : http://research-pub.gene.com/gmap/                 
  10. deBGA :  https://github.com/hitbc/deBGA                      
                                                      
## IV. Check quality for BAM files
                     
## V. Filter and correct the mapped reads
                         
## VI. Call peaks and differential peaks
                     
## VII. Conversion, Normalization and Reads density calculation
                                  
## VIII. Peaks annotation   
                                        
