#Top 10 softwares for ChIP-seq data analysis. (The 2nd generation sequencing)
             
## I. Check quality for FASTQ files
  1. FastQC : http://www.bioinformatics.babraham.ac.uk/projects/fastqc/              
  2. FastQ Screen : http://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/             
  3. MultiQC : http://multiqc.info/    
  4. NGSUtils : http://ngsutils.org/          
  5. ngs-bits : https://github.com/imgag/ngs-bits                   
  6. fastqp : https://pypi.python.org/pypi/fastqp            
  7. QC3 : https://github.com/slzhao/QC3                      
  8. Rqc : https://bioconductor.org/packages/release/bioc/html/Rqc.html               
  9. ShortRead : https://bioconductor.org/packages/release/bioc/html/ShortRead.html              
  10. seqTools : http://www.bioconductor.org/packages/release/bioc/html/seqTools.html              
                          
## II. Filter and correct the raw reads
  1. Trimmomatic : http://www.usadellab.org/cms/?page=trimmomatic     
  2. cutadapt ï¼š https://pypi.python.org/pypi/cutadapt     
  3. Galore : http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/         
  4. AdapterRemoval :  https://github.com/MikkelSchubert/adapterremoval              
  5. 
                  
## III. Reads mapping (Mappers or Aligners)
  1. BWA-mem and BWA-aln:    https://github.com/lh3/bwa        
  2. Bowtie2 :      https://github.com/BenLangmead/bowtie2   
  3. Stampy :   www.well.ox.ac.uk/bioinformatics/Software/Stampy-latest.tgz   
  4. Novoalign : http://www.novocraft.com/support/download/  
  5. Subread : http://subread.sourceforge.net/     
  6. Yara : https://www.seqan.de/apps/yara/       
     RazerS 3 : https://www.seqan.de/apps/razers-3/      
  7. mrsFAST-Ultra :  http://sfu-compbio.github.io/mrsfast/      
  8. BBMap : https://sourceforge.net/projects/bbmap/            
  9. GMAP-GSNAP : http://research-pub.gene.com/gmap/                 
  10. deBGA :  https://github.com/hitbc/deBGA                      
                                                      
## IV. Check quality for BAM files
                     
## V. Filter and correct the mapped reads
                         
## VI. Call peaks and differential peaks
                     
## VII. Conversion, Normalization and Reads density calculation
                                  
## VIII. Peaks annotation   
                                        
