# Top 10 softwares for RNA-seq data analysis.                    
####  Only for the 2nd generation sequencing of bulk cell populations.
##### Softwares for Single-cell RNA-seq data analysis, please see :                         
##### https://github.com/CTLife/Top10/blob/master/Single-cell_RNA-seq.md   
                                 
-----------------------                          
                                                             
## I. Check quality for FASTQ files
  1. FastQC : http://www.bioinformatics.babraham.ac.uk/projects/fastqc/              
  2. FastQ Screen : http://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/                  
  3. NGS QC Toolkit : http://www.nipgr.res.in/ngsqctoolkit.html                               
  4. fastq-tools : https://github.com/dcjones/fastq-tools                                  
  5. fastqp : https://pypi.python.org/pypi/fastqp            
  6. QC3 : https://github.com/slzhao/QC3                      
  7. PRINSEQ : http://prinseq.sourceforge.net/              
  8. Rqc : https://bioconductor.org/packages/release/bioc/html/Rqc.html                                           
  9. seqTools : http://www.bioconductor.org/packages/release/bioc/html/seqTools.html     
  10. ShortRead : http://bioconductor.org/packages/release/bioc/html/ShortRead.html                                              
  MultiQC : https://github.com/ewels/MultiQC   (Aggregate the results from FastQC or FastQ Screen analyses across many samples into a single report.)             
  Other tools : [fastools](https://pypi.python.org/pypi/fastools) , [fqtools](https://github.com/alastair-droop/fqtools)  ,  [NGSUtils](http://ngsutils.org/) , [ngs-bits](https://github.com/imgag/ngs-bits) , [SolexaQA](http://solexaqa.sourceforge.net/)  , [systemPipeR](https://bioconductor.org/packages/release/bioc/html/systemPipeR.html)                                                                     
                   
-------------------------
             
## II. Filter and correct the raw reads
  1. Trimmomatic : http://www.usadellab.org/cms/?page=trimmomatic     
  2. cutadapt ï¼š https://pypi.python.org/pypi/cutadapt     
  3. Galore : http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/         
  4. AdapterRemoval :  https://github.com/MikkelSchubert/adapterremoval              
  5. Flexbar : https://github.com/seqan/flexbar 
  6. SeqPrep : https://github.com/jstjohn/SeqPrep               
  7. SeqPurge : https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1069-7                   
  8. Lighter : https://github.com/mourisl/Lighter        
  9. Karect : http://aminallam.github.io/karect/          
  10. CoLoRMap : https://github.com/sfu-compbio/colormap                       
                                                                                                      
-------------------------
                                                                              
## III. Reads mapping (Mappers or Aligners)
  1. HISAT2: https://ccb.jhu.edu/software/hisat2/index.shtml         
  2. STAR: https://github.com/alexdobin/STAR                    
  3. RSEM: http://deweylab.github.io/RSEM/                 
     Kallisto: http://pachterlab.github.io/kallisto/                    
     Alpine: http://bioconductor.org/packages/release/bioc/html/alpine.html                   
  4. Sailfish: https://github.com/kingsfordgroup/sailfish                
     Salmon: https://combine-lab.github.io/salmon/                 
     RapMap: https://github.com/COMBINE-lab/RapMap                     
  5. MapSplice: http://www.netlab.uky.edu/p/bioinfo/MapSplice2                       
  6. CRAC: http://crac.gforge.inria.fr/               
  7. Subjunc: http://subread.sourceforge.net/               
  8. BBMap: https://sourceforge.net/projects/bbmap/       
  9. Novoalign: http://www.novocraft.com/products/novoalign/     
     GSNAP: http://research-pub.gene.com/gmap/    
  10. FBB:  https://bitbucket.org/irenerodriguez/fbb     
      TACO:  https://tacorna.github.io/             
      CLC Genomics Workbench: https://www.qiagenbioinformatics.com/products/clc-genomics-workbench        
                                          
-------------------------
                                              
## IV. Check quality for BAM files
1. Sambamba :  https://github.com/lomereiter/sambamba   
2. Biobambam2 : https://github.com/gt1/biobambam2  
3. NGS-QC Generator : http://www.ngs-qc.org/                            
4. htSeqTools : http://bioconductor.org/packages/release/bioc/html/htSeqTools.html            
5. BatchQC : http://bioconductor.org/packages/release/bioc/html/BatchQC.html                 
                                          
-------------------------
                                 
## V. Filter and correct the mapped reads
                         
   
                                        
